app/controllers/trainees/placements/confirm_details_controller.rb

```diff
@@ -3,13 +3,21 @@
 module Trainees
   module Placements
     class ConfirmDetailsController < Trainees::ConfirmDetailsController
+      include TraineeHelper
+
       def update
         trainee.draft? ? toggle_trainee_progress_field : @form.save!
 
         flash[:success] = "Trainee #{flash_message_title} updated"
 
         redirect_to(page_tracker.last_origin_page_path || trainee_path(trainee))
       end
+
+    private
+
+      def authorize_trainee
+        placement_editable?(trainee)
```

Comment:
i'd marginally prefer just inlining the `policy(trainee).write_placements?` and not bothering with the helper method